src/topology/metric_space/emetric_space.lean

```diff
@@ -590,11 +590,11 @@ mem_of_superset (ball_mem_nhds x ε0) ball_subset_closed_ball
 
 theorem ball_prod_same [pseudo_emetric_space β] (x : α) (y : β) (r : ℝ≥0∞) :
   ball x r ×ˢ ball y r = ball (x, y) r :=
-ext $ λ z, max_lt_iff.symm
+ext $ λ z, (@max_lt_iff ℝ≥0∞ _ _ _ _).symm
```

Comment:
I have no idea why this broke.

Reply:
This is extremely mysterious. What was the error you got?

Reply:
Something about the lemma (with a bunch of metavariables) not matching what was expected. I've come to expect this sort of random breakage from any typeclass refactors, but this one is particularly mysterious because it seems to have nothing to do with the PR.

Reply:
Here's my guess: max_lt_iff uses `linear_order` and we're on `nnreal`. `nnreal` doesn't appear to have an instance of `linear_order` directly: it has `conditionally_complete_linear_order_bot` and `canonically_linear_ordered_add_monoid`, both of whom give an instance of `linear_order` (and the former might be touched by the refactor?)

If this is a diamond of some kind, and it might be, then perhaps the refactor has made something not def eq there?

That's all I've got. Probably worth investigating though.